server/src/internalClusterTest/java/org/elasticsearch/action/admin/indices/create/AutoCreateSystemIndexIT.java

```diff
@@ -167,40 +168,60 @@ public void testSystemIndicesAutoCreateRejectedWhenNotHidden() {
         );
     }
 
-    /**
-     * Check that a template applying a system alias creates a hidden alias.
-     */
-    public void testAutoCreateSystemAliasViaV1Template() throws Exception {
+    private String autoCreateSystemAliasViaV1Template(String indexName) throws Exception {
         assertAcked(
             client().admin()
                 .indices()
                 .preparePutTemplate("test-template")
-                .setPatterns(List.of(INDEX_NAME + "*"))
-                .addAlias(new Alias(INDEX_NAME + "-legacy-alias"))
+                .setPatterns(List.of(indexName + "*"))
+                .addAlias(new Alias(indexName + "-legacy-alias"))
                 .get()
         );
 
-        String nonPrimaryIndex = INDEX_NAME + "-2";
+        String nonPrimaryIndex = indexName + "-2";
         CreateIndexRequest request = new CreateIndexRequest(nonPrimaryIndex);
         assertAcked(client().execute(AutoCreateAction.INSTANCE, request).get());
-
         assertTrue(indexExists(nonPrimaryIndex));
 
-        assertAliasesHidden(nonPrimaryIndex, Set.of(".test-index", ".test-index-legacy-alias"));
+        return nonPrimaryIndex;
+    }
+
+    /**
+     * Check that a legacy template does not create an alias for a system index
+     */
+    public void testAutoCreateSystemAliasViaV1Template() throws Exception {
+        var nonPrimaryIndex = autoCreateSystemAliasViaV1Template(INDEX_NAME);
+
+        assertAliasesHidden(nonPrimaryIndex, Set.of(INDEX_NAME), 1);
 
         assertAcked(client().admin().indices().prepareDeleteTemplate("*").get());
     }
 
     /**
-     * Check that a composable template applying a system alias creates a hidden alias.
+     * Check that a legacy template does create an alias for a system index, because of allows templates
      */
-    public void testAutoCreateSystemAliasViaComposableTemplate() throws Exception {
+    public void testAutoCreateSystemAliasViaV1TemplateAllowsTemplates() throws Exception {
+        var nonPrimaryIndex = autoCreateSystemAliasViaV1Template(TestSystemIndexDescriptorAllowsTemplates.INDEX_NAME);
+
+        assertAliasesHidden(
+            nonPrimaryIndex,
+            Set.of(
+                TestSystemIndexDescriptorAllowsTemplates.INDEX_NAME,
+                TestSystemIndexDescriptorAllowsTemplates.INDEX_NAME + "-legacy-alias"
+            ),
+            2
+        );
+
+        assertAcked(client().admin().indices().prepareDeleteTemplate("*").get());
```

Comment:
This line isn't necessary, it's done in the `@After` (and this should be handled by the test infrastructure anyway, I thought).